src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -548,19 +552,43 @@ private void VerifyConnected()
 			Func<object, X509Certificate, X509Chain, SslPolicyErrors, bool> remoteCertificateCb =
 				(rcbSender, rcbCertificate, rcbChain, rcbPolicyErrors) =>
 				{
-					switch (rcbPolicyErrors)
+					if (cs.SslMode == MySqlSslMode.Preferred || cs.SslMode == MySqlSslMode.Required)
+						return true;
+
+					if ((rcbPolicyErrors & SslPolicyErrors.RemoteCertificateChainErrors) > 0 && cs.CACertificate != null)
```

Comment:
Nitpick: I would write `!= 0` for testing a bitflag. (`> 0` first made me think it was a count of errors, not testing a flag.)